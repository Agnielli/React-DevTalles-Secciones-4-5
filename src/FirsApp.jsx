import PropTypes from 'prop-types';

const newMessage = {
  name: 'Enrique',
  message: 'Hola Mundo',
};

const getResult = (a, b) => a + b


export const FirsApp = ({
  title, 
  subTitle, 
}) => {
  
  const { message, name } = newMessage

  if (!title) {
    throw new Error('No existe el titulo')
  }

  return (
    <>
      <h1>{ name }</h1>
      <h2>{ title }</h2>
      <h3>{ subTitle }</h3>
      {/* <code>{ JSON.stringify( newMessage ) }</code> */}

      <p>{ getResult(6, 4) }</p>
      <p>{ message }</p>
    </>
  )
} 

FirsApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
}

FirsApp.defaultProps = {
  title: 'NO HAY TITLE',
  subTitle: 'No hay subtitulo',
  name: 'No name',
}
