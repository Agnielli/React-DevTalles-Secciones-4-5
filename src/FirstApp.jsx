import PropTypes from 'prop-types';

const newMessage = {
  name: 'Enrique',
  message: 'Hola Mundo',
};

const getResult = (a, b) => a + b


export const FirstApp = ({
  title, 
  subTitle, 
}) => {
  
  const { message, name } = newMessage

  if (!title) {
    throw new Error('No existe el titulo')
  }

  return (
    <>
      <h1>{ title }</h1>
      <h3>{ subTitle }</h3>
      {/* <code>{ JSON.stringify( newMessage ) }</code> */}

      <p>{ getResult(6, 4) }</p>
      <p>{ message }</p>
    </>
  )
} 

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
  title: 'Enrique Sabariego',
  subTitle: 'No hay subtitulo',
  name: 'No name',
}
