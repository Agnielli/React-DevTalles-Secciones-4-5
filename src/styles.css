

html, body {
  background-color: #21232A;
  color: white;
  font-family: Helvetica, Arial, sans-serif;
  padding: 1.3rem;
}

button {
  padding: 15px;
  font-size: 1.3rem;
  margin-right: 5px;
}