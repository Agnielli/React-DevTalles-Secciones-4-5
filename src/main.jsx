import React from "react";
import ReactDOM from "react-dom/client";
// import App from "./App.jsx";
// import { FirsApp } from "./FirsApp.jsx";
import { CounterApp } from "./CounterApp";
import './styles.css'

ReactDOM.createRoot( document.getElementById('root') ).render(
  <React.StrictMode>
    {/* <App /> */}
    {/* <FirsApp  subTitle = 'La tierra explota'/> */}
    <CounterApp value={20} />
  </React.StrictMode>
)