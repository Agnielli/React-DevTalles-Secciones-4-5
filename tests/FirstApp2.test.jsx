/**
 * @jest-environment jsdom
 */
import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en FirstApp', () => {

  const title = 'Soy kike';
  const subTitle = 'Soy un subTitle';

  test('debe hacer match con el snapshot', () => {
    const { container } = render( <FirstApp title = 'title' /> );
    expect( container ).toMatchSnapshot()
  });  

  test('debe de mostrar el mensaje "Soy kike"', () =>{
    
    render( < FirstApp title={title} /> );
    expect( screen.getByText(title) ).toBeTruthy();
    //screen.debug();
  });

  test('debe mostrar el título en un h1', () =>{
    render( < FirstApp title={title} /> );
    expect( screen.getByRole('heading', { level:1 }).innerHTML ).toContain( title );
  });

  test('debe de mostrar el subtítulo enviado por props', () => {
    render( < FirstApp 
                title = {title}
                subTitle = {subTitle} 
            />
          );

    expect( screen.getAllByText(subTitle).length ).toBe(2);
  })
});