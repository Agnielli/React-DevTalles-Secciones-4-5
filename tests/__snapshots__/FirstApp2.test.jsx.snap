// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    title
  </h1>
  <h3>
    No hay subtitulo
  </h3>
  <h3>
    No hay subtitulo
  </h3>
  <p>
    10
  </p>
  <p>
    Hola Mundo
  </p>
</div>
`;
